packages/app/components/token-picker/TokenPicker.tsx

```diff
@@ -1,69 +1,105 @@
+
+interface TokenListRowProps {
+  token: Token;
+  closeAction: () => void;
+}
+
+interface TokenListProps {
+  closeAction: () => void;
+  tokenList: Token[];
+}
 
 export const TokenPicker = ({ isOpen, closeAction }: ModalBaseProps) => {
+  const { tokenList } = useTokenListContext();
+
   return (
-    <div>
-      <Modal isOpen={isOpen} closeAction={closeAction}>
-        <ModalHeaderTitle title="Select a token" closeAction={closeAction} />
-        <ModalContent>
-          <div className="space-y-4">
-            <input
-              type="search"
-              placeholder="search token name or paste address"
-              className="w-full p-2 bg-surface-75 rounded-xl"
-            />
-            <div className="py-3 space-y-2">
-              <p className="text-xs font-semibold text-em-low">Common tokens</p>
-              <div className="flex items-center space-x-3">
-                <div className="p-2 font-semibold uppercase border text-em-med w-fit rounded-2xl">
-                  Eth
-                </div>
-                <div className="p-2 font-semibold uppercase border text-em-med w-fit rounded-2xl">
-                  Pepe
-                </div>
-                <div className="p-2 font-semibold uppercase border text-em-med w-fit rounded-2xl">
-                  usdc
-                </div>
-              </div>
-            </div>
-            <div className="relative">
-              <div className="-mx-4 overflow-y-auto border-t h-72 border-surface-50">
-                {[
-                  "1Inch",
-                  "Aave",
-                  "DAI",
-                  "USDC",
-                  "Matic",
-                  "Uni",
-                  "pepe",
-                  "Swapr",
-                  "ARB",
-                ].map((token) => (
-                  <div
-                    className="flex justify-between w-full px-4 py-2 border-b cursor-pointer border-surface-50 hover:bg-surface-50"
-                    key={token}
-                    onClick={closeAction}
-                  >
-                    <div className="flex items-center space-x-3">
-                      <div className="w-8 h-8 rounded-full bg-primary-100" />
-                      <BodyText>{token}</BodyText>
-                    </div>
-                    <BodyText>0</BodyText>
-                  </div>
-                ))}
-              </div>
-            </div>
-          </div>
-        </ModalContent>
-      </Modal>
-    </div>
+    <Modal closeAction={closeAction} isOpen={isOpen}>
+      <ModalHeaderTitle closeAction={closeAction} title="Select a token" />
+      <ModalContent>
+        <SearchBar />
+        <CommonTokens />
+        <TokenList closeAction={closeAction} tokenList={tokenList} />
+      </ModalContent>
+    </Modal>
   );
 };
+
+const SearchBar = () => (
+  <div className="flex items-center bg-surface-50 border border-surface-75 rounded-xl py-2 px-3 text-em-low mx-6">
```

Comment:
I think we can remove the padding x from elements.

```suggestion
  <div className="flex items-center bg-surface-50 border border-surface-75 rounded-xl py-2 px-3 text-em-low">
```